lib/improver/tests/ensemble_calibration/ensemble_calibration_utilities/test_SplitHistoricForecastAndTruth.py

```diff
@@ -0,0 +1,223 @@
+            "truth_dict={'attributes': "
+            "{'mosg__model_configuration': 'uk_det'}}>")
+        self.assertEqual(result, expected_result)
+
+
+class Test__find_required_cubes_using_metadata(SetupCubesAndDicts):
+    """Test the _find_required_cubes_using_metadata method."""
+
+    def test_attributes(self):
+        """Test that the desired cube is returned if the required attributes
+        are specified."""
+        result = self.plugin._find_required_cubes_using_metadata(
+            self.combined, self.historic_forecast_dict)
+        self.assertEqual(result, self.historic_forecasts)
+
+    def test_non_attributes(self):
+        """Test that the if metadata other than attributes are specified then
+        a NotImplementedError is raised."""
+        input_dict = {
+            "coord": "forecast_period==0"
+        }
+        msg = "only constraining on attributes is supported"
+        with self.assertRaisesRegex(NotImplementedError, msg):
+            self.plugin._find_required_cubes_using_metadata(
+                self.combined, input_dict)
+
+    def test_non_matching_attributes(self):
+        """Test that the if attributes are specified but these do not match
+        any attributes on the cube then a ValueError is raised."""
+        input_dict = {
+            "attributes": {
+                "mosg__model_configuration": "mars_det"
+            }
+        }
+        msg = "The metadata to identify"
+        with self.assertRaisesRegex(ValueError, msg):
+            self.plugin._find_required_cubes_using_metadata(
+                self.combined, input_dict)
+
+
+class Test__filter_non_matching_cubes(SetupCubesAndDicts):
+    """Test the _filter_non_matching_cubes method."""
+
+    def test_all_matching(self):
+        """Test for when the historic forecast and truth cubes all match."""
+        hf_result, truth_result = self.plugin._filter_non_matching_cubes(
+            self.historic_forecasts, self.truth)
+        self.assertEqual(hf_result, self.historic_forecasts)
+        self.assertEqual(truth_result, self.truth)
+
+    def test_fewer_historic_forecasts(self):
+        """Test for when there are fewer historic forecasts than truths,
+        for example, if there is a missing forecast cycle."""
+        hf_result, truth_result = self.plugin._filter_non_matching_cubes(
+            self.partial_historic_forecasts, self.truth)
+        self.assertEqual(hf_result, self.partial_historic_forecasts)
+        self.assertEqual(truth_result, self.partial_truth)
+
+    def test_fewer_truths(self):
+        """Test for when there are fewer truths than historic forecasts,
+        for example, if there is a missing analysis."""
+        hf_result, truth_result = self.plugin._filter_non_matching_cubes(
+            self.historic_forecasts, self.partial_truth)
+        self.assertEqual(hf_result, self.partial_historic_forecasts)
+        self.assertEqual(truth_result, self.partial_truth)
+
```

Comment:
In line with my comment about the call to this function, could you also have a test where they are mismatched, but differently so (e.g. one missing the first validity time, and one the last, resulting in a list that is 2 elements shorter than the total possible number).

Reply:
I've added a test that is similar to what you've described.